uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -266,6 +266,22 @@ define([
       };
 
       /**
+       * Checks if the leave request is of a specific status(sent as a param)
+       * If 'statusBeforeEdit' is available ->its an existing request, so search is done on this object
+       * If 'statusBeforeEdit' is not present -> its a new request, so search is done on this.request object
+       *
+       * @param {String} leaveStatus
+       * @return {Boolean}
+       */
+      this.isLeaveStatus = function (leaveStatus) {
+        var status = this.statusBeforeEdit.name
+          ? this.statusBeforeEdit
+          : getStatusFromValue.call(this, this.request.status_id);
+
+        return status ? status.name === leaveStatus : false;
```

Comment:
why is simply getting the leave request status is so complicated? it should be something like this
```js
this.isStatus = function (statusName) {
  var statusId = // using the statusName, get the id

  return this.request.status_id === statusId;
}
```